index.d.ts

```diff
@@ -26,6 +26,8 @@ export interface ThunkDispatch<
     thunkAction: ThunkAction<TReturnType, TState, TExtraThunkArg, TBasicAction>
   ): TReturnType;
   <A extends TBasicAction>(action: A): A;
+  // This overload is the union of the two above (see TS issue #14107).
+  <R, T extends A>(action: T | ThunkAction<R, S, E, A>): T | R;
```

Comment:
I think you copied this in without checking for the updated variable names. Here it is formatted and fixed:
```suggestion
  <TReturnType, TAction extends TBasicAction>(
    action:
      | TAction
      | ThunkAction<TReturnType, TState, TExtraThunkArg, TBasicAction>
  ): TAction | TReturnType;
```

Reply:
Right, sorry about that.